# m

In [1]:
print("helllo")

helllo


In [2]:
import string,time


'!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'

In [6]:
# total number of punctutaion in python
total_punctuations=string.punctuation
total_punctuations

'!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'

In [7]:
def remove_punc(text):
    for char in total_punctuations:
        text=text.replace(char,'')
    return text

In [8]:
text='string. with. punctuation?'

In [23]:
#time it will for maunal function to remove punctuation
start_time=time.time()
print(remove_punc(text))
finishing_time=time.time()
time_taken_by_manual=finishing_time-start_time
print(time_taken_by_manual*100000)

string with punctuation
30.040740966796875


# automatic removal of punctation mark using lib

In [18]:
def punc_auto(text):
    return text.translate(str.maketrans('','',total_punctuations))


In [24]:
#time it will for maunal function to remove punctuation
start_time=time.time()
print(punc_auto(text))
finishing_time=time.time()
time_taken_by_auto=finishing_time-start_time
print(time_taken_by_auto*100000)

string with punctuation
36.47804260253906


In [26]:
# time comparision between how mast is manual and autp
print((time_taken_by_auto/time_taken_by_manual)*100)

121.42857142857142


# conclusion 
    # auto is 121 percent faster than manual removal 
    of punctuation marks